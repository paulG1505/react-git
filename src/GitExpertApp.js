//rafc sinnipet
import React, { useState } from 'react'
import { CategoryAdd } from './components/CategoryAdd'
import { GifGrid } from './components/GifGrid'
//le mandamos unas categorias por defecto
export const GitExpertApp = ({defaultCategories=[]}) => {
    const [categories, setCategories] = useState(defaultCategories)
    

    return (
        <div>
            <h2>GifExpertApp</h2>
            <CategoryAdd setCategories={setCategories} />
            <hr />

            <ol>
                {/* recorrer el arreglo con map  recibe obj y 
                lo ponemos como key*/}
                {
                    categories.map(category => (
                        <GifGrid 
                        key={category}
                        category={category} />
                    ))
                }
            </ol>
        </div>
    )
}

export default GitExpertApp
