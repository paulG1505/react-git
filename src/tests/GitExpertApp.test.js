import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import React from 'react'
import GitExpertApp from '../GitExpertApp';

describe('Pruebas a GitExpertApp', () => {
    test('Renderizado el componente con snapshot', () => {
        const wrapper=shallow(<GitExpertApp/>)
        expect(wrapper).toMatchSnapshot()
    })
    test('Comprobamos que existan categorias iniciales', () => {
        const categories=['Homer']
        const wrapper=shallow(<GitExpertApp defaultCategories={categories}/>)
        expect(wrapper).toMatchSnapshot()//comprobamos que exista
        //comprobamos que el initialState
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    })
    
})
