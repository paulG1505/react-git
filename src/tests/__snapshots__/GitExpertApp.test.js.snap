// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a GitExpertApp Comprobamos que existan categorias iniciales 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Homer"
      key="Homer"
    />
  </ol>
</div>
`;

exports[`Pruebas a GitExpertApp Renderizado el componente con snapshot 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
